### PR TITLE
Truncate Component

### PR DESCRIPTION
**Pull request by BeniCheni**
Per #58 , this PR implements `Truncate` component to support syntax such as:
```
<Truncate>
  {loripsum}
</Truncate>
```

##### Story - Without Container
screen shot 2017-10-03 at 4 03 10 pm

##### Story - With Container
screen shot 2017-10-03 at 4 03 23 pm

##### Unit Test Status
```
design-system> npm test -- -u

> pcln-design-system@1.0.0-9 test /Users/bchen/priceline-web/design-system
> jest "-u"

 PASS  src/__tests__/Truncate.js
 PASS  src/__tests__/Icon.js
 PASS  src/__tests__/Text.js
 PASS  src/__tests__/Heading.js
 PASS  src/__tests__/Button.js
 PASS  src/__tests__/Box.js
 PASS  src/__tests__/Flex.js
 PASS  src/__tests__/Link.js
 PASS  src/__tests__/ThemeProvider.js
 PASS  src/__tests__/OutlineButton.js
 PASS  src/__tests__/RedButton.js
 PASS  src/__tests__/Image.js
 PASS  src/__tests__/GreenButton.js
 PASS  src/__tests__/BackgroundImage.js
 PASS  src/__tests__/theme.js

Test Suites: 15 passed, 15 total
Tests:       97 passed, 97 total
Snapshots:   96 passed, 96 total
Time:        3.997s
Ran all test suites.

=============================== Coverage summary ===============================
Statements   : 100% ( 160/160 )
Branches     : 100% ( 18/18 )
Functions    : 100% ( 22/22 )
Lines        : 100% ( 150/150 )
================================================================================
```

### CODE REVIEW

**Comment by broox9** [comment]
+1

File: src/Truncate.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Text from './Text'
+
+const Truncate = Text.extend`
```
Comment (jxnblk) [inline src/Truncate.js]: 👍

File: .storybook/Truncate.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Duo Reges: constructio interrete. Mihi enim satis est, ipsis non satis. Si enim ita est, vide ne facinus facias, cum mori suadeas. Illud dico, ea, quae dicat, praeclare inter se cohaerere. Est enim tanti philosophi tamque nobilis audacter sua decreta defendere.
+`
+
+storiesOf('Truncate', module)
+  .add('Without Container', () => (
+    <Truncate width={50}>{loripsum}</Truncate>
+  ))
+  .add('With Container', () => (
+    <Box width='30%'>
```
Comment (jxnblk) [inline .storybook/Truncate.js]: This can also be a fraction, e.g. `width={3/10}`

Comment (BeniCheni) [reply .storybook/Truncate.js]: 👍 Updated.

**Comment by jxnblk** [comment]
Something funky is going on with travis, and it keeps seeming to fail for reasons unrelated to the PRs...

**Pull request merged**